manifests/init.pp

```diff
@@ -13,102 +13,90 @@
+  $logroot_mode                                                  = $::apache::params::logroot_mode,
+  $log_level                                                     = $::apache::params::log_level,
+  $log_formats                                                   = {},
+  $ssl_file                                                      = $::apache::params::ssl_file,
+  $ports_file                                                    = $::apache::params::ports_file,
+  $docroot                                                       = $::apache::params::docroot,
+  $apache_version                                                = $::apache::version::default,
+  $server_tokens                                                 = 'OS',
+  $server_signature                                              = 'On',
+  $trace_enable                                                  = 'On',
+  Optional[Enum['on', 'off', 'nodecode']] $allow_encoded_slashes = undef,
+  $file_e_tag                                                    = undef,
+  $package_ensure                                                = 'installed',
+  Boolean $use_optional_includes                                 = $::apache::params::use_optional_includes,
+  $use_systemd                                                   = $::apache::params::use_systemd,
+  $mime_types_additional                                         = $::apache::params::mime_types_additional,
+  $file_mode                                                     = $::apache::params::file_mode,
+  $root_directory_options                                        = $::apache::params::root_directory_options,
+  Boolean $root_directory_secured                                = false,
+  $error_log                                                     = $::apache::params::error_log,
+  $scriptalias                                                   = $::apache::params::scriptalias,
+  $access_log_file                                               = $::apache::params::access_log_file,
 ) inherits ::apache::params {
-  validate_bool($default_vhost)
-  validate_bool($default_ssl_vhost)
-  validate_bool($default_confd_files)
-  # true/false is sufficient for both ensure and enable
-  validate_bool($service_enable)
-  validate_bool($service_manage)
-  validate_bool($use_optional_includes)
-  validate_bool($root_directory_secured)
 
   $valid_mpms_re = $apache_version ? {
     '2.4'   => '(event|itk|peruser|prefork|worker)',
     default => '(event|itk|prefork|worker)'
   }
 
   if $mpm_module and $mpm_module != 'false' { # lint:ignore:quoted_booleans
-    validate_re($mpm_module, $valid_mpms_re)
-  }
-
-  if $allow_encoded_slashes {
-    validate_re($allow_encoded_slashes, '(^on$|^off$|^nodecode$)', "${allow_encoded_slashes} is not permitted for allow_encoded_slashes. Allowed values are 'on', 'off' or 'nodecode'.")
+    assert_type(Pattern[$valid_mpms_re], $mpm_module)
```

Comment:
I wonder if this could be rewritten to an Enum. That's most likely an easier to read error.